### PR TITLE
Use swr mutate for addresses

### PR DESCRIPTION
**Pull request by Meyclem**
## Description

This PR adds the mutation of SWR's cache when updating or creating an address.
I did the same for "mark as primary" and "delete" actions.

## Type of change

- [ ] **Chore** (non-breaking change which refactors / improves the existing code base).
- [ ] **Bug fix** (non-breaking change which fixes an issue).
- [x] **New feature** (non-breaking change which adds functionality).
- [ ] **Breaking change** (fix or feature that would cause existing functionality to not work as expected).

## Issues

[CU-adzn4p](https://app.clickup.com/t/adzn4p)

## Checklist

- [x] My code follows the style [**contributing guidelines**][contributing_file] of this project.
- [x] I have performed a self-review of my own code.
- [ ] I have commented my code in hard-to-understand areas.
- [ ] I have made corresponding changes to the documentation.
- [x] My changes generate no new warnings.
- [ ] I have added tests that prove my fix is effective or that my feature works.

[contributing_file]: https://github.com/fewlinesco/connect-account/blob/master/README.adoc

### CODE REVIEW

**Comment by frieda-ewlines** [comment]
Task linked: [CU-adzn4p Use `mutate` instead of `cache` from SWR for addresses creation and update](https://app.clickup.com/t/adzn4p)

File: src/components/forms/profile/user-address-form.tsx [hunk 1]
```diff
@@ -94,6 +116,30 @@ const UserAddressForm: React.FC<{
     street_address_2: "",
     primary: false,
   });
+  const [addresses, setAddresses] = React.useState<Address[]>([]);
```
Comment (Meyclem) [inline src/components/forms/profile/user-address-form.tsx:119]: We need to use that to get the addresses. If we don't, we don't have any other way to get the value of the cache.

File: src/components/pages/address-overview/address-overview.tsx [hunk 2]
```diff
@@ -130,7 +185,21 @@ const AddressOverview: React.FC<{ address?: Address }> = ({ address }) => {
             variant={ButtonVariant.PRIMARY}
             onPress={() => {
               fetchJson(`/api/profile/addresses/${address?.id}`, "DELETE", {})
-                .then(() => router && router.push("/account/profile"))
+                .then(async () => {
+                  router && router.push("/account/profile");
```
Comment (Oluane) [inline src/components/pages/address-overview/address-overview.tsx:189]: It doesn't change anything to have the `router.push()` before the rest of the statements ?

Comment (Meyclem) [reply src/components/pages/address-overview/address-overview.tsx:189]: You're right, it works but it's not clean and it make some things happen after the page changed.

I'm changing that 👍

Comment (Oluane) [reply src/components/pages/address-overview/address-overview.tsx]: Alright, that was my point, I wanted to know first (for curiosity sake) if it worked smoothly like that. It's good to know it. 🙂 
But indeed, I find it cleaner too to have this piece as the last statement 👌

**Review by Oluane — DISMISSED** [review]
Good for me. Nice work 👍

**Review by Fuzznimp — DISMISSED** [review]

**Review by Fuzznimp — APPROVED** [review]

**Pull request merged**